js/index.js

```diff
@@ -17,3 +17,125 @@ for (let i = 0; i < skills.length; i++) {
+
+messageSection.hide = function () {
+    this.style.display = 'none';
+};
+messageSection.show = function () {
+    this.style.display = '';
+};
+
+Object.defineProperty(messageList, "hasNoMessages", {
+    get: function () {
+        return this.childElementCount === 0
+    }
+});
+
+function updateMessageSection() {
+    if (messageList.hasNoMessages) {
+        messageSection.hide();
+    }
+};
+updateMessageSection();
+
+
+messageForm.addEventListener('submit', (e) => {
+    e.preventDefault();
+    const form = e.target;
+
+    const userName = form.usersName.value;
+    const useEmail = form.usersEmail.value;
+    const userMessage = form.usersMessage.value;
+
+    // console.log(userName);
+    // console.log(useEmail);
+    // console.log(userMessage);
+
+    const newMessage = document.createElement('li');
+    newMessage.innerHTML = `
+        <a href="mailto:${useEmail}" target="_blank">${userName}</a>
+        wrote: <span> ${userMessage}</span>
+    `;
+    newMessage.messageText = userMessage;
+
+    const removeButton = document.createElement('button');
+    removeButton.innerText = "remove";
+    removeButton.type = "button";
+
+    const editButton = document.createElement('button');
+    editButton.innerText = "edit";
+    editButton.type = "button";
+
+
+    newMessage.appendChild(editButton);
+    newMessage.appendChild(removeButton);
+    messageList.appendChild(newMessage);
+    messageSection.show();
+
+    form.reset();
+});
+
+messageList.addEventListener('click', (e) => {
+    if (e.target.tagName === 'BUTTON') {
+        const button = e.target;
+        const listButtons = messageList.querySelectorAll('button');
+        const li = button.parentNode;
+        const messageText = li.messageText;
+        const action = button.textContent;
+
+        const disableButton = (btn) => {
+            if (btn.parentNode !== li) {
+                btn.disabled = true;
+            }
+        };
+        const enableButton = (btn) => btn.disabled = false;
+        const switchButtons = (buttons, switcher) => {
+            buttons.forEach(button => switcher(button));
+        }
+
+        const nameActions = {
+            edit: () => {
+                switchButtons(listButtons, disableButton);
+                const span = li.querySelector('span');
+                const input = document.createElement('input');
+                input.type = "text";
+                input.value = span.textContent;
+                li.insertBefore(input, span);
+                li.removeChild(span);
+                button.textContent = 'save';
+            },
+            save: () => {
+                const input = li.querySelector('input');
+                const span = document.createElement('span');
+                const editedMessage = input.value.trim();
+
+                if (editedMessage === '') {
+                    const answer = confirm('Delete message?');
+                    if (answer === true) {
```

Comment:
tiny nit: because `answer` is already assigned to a boolean value, you can use the `answer` as the entire conditional. imo this wouldn't cost any readability
```suggestion
                    if (answer) {
```

Reply:
I've just made it in vs code